### ISSUE TITLE
Quality check on specific `ref_ids`

### ISSUE THREAD

**Issue by kvantricht** [WorldCereal/presto-worldcereal#23]
- [ ] 2018_ES_SIGPAC-Andalucia
- [ ] 2019_ES_SIGPAC-Andalucia
- [x] 2021_AT_LPIS
- [x] 2021_LV_LPIS

Detailed scores on how performance drops once the region (not the ref_id, but all coutry-years) is removed from training can be found [here](https://vitoresearch-my.sharepoint.com/:x:/g/personal/christina_butsko_vito_be/EYEvsyl8lllGpNiCc234XnYBwhoYQKJD2AvoUXViwFe4GA?e=0KA9vG)

**Comment by CButsko** [comment]
- [ ] run exhaustive loop on all ref_ids to see the whole picture
- [ ] for those ref_ids that demonstrate significant but not radical drop of quality when removed from training (~20% drop) - need to plot distrubution of labels (idea: these ref_ids might have concentration or "tricky" grass-like labels)

**Comment by cbutsko** [comment]
[This table](https://vitoresearch-my.sharepoint.com/:x:/g/personal/christina_butsko_vito_be/EYEvsyl8lllGpNiCc234XnYBwhoYQKJD2AvoUXViwFe4GA?e=MqwXAj) shows results for performance on _ref_id-country_ level where none of the country data was used in training. In places where multiple overlapping datasets exist, this can be a good way to see the quality. In particular, the following LPIS datasets show worse performance compared to 2021_EUR_EXTRACROPS or LUCAS:
- [ ] 2019_LV_LPIS and 2021_LV_LPIS
- [ ] all Austrian LPIS
- [ ] Belgium (particularly 2018)

Other datasets likely to have bad labels:
- [ ] 2020_GLO_ewoco_* in Ukraine
- [ ] 2020_GLO_ewoco_* in Russia
- [ ] 2020_GLO_ewoco_* in Brazil
- [ ] 2020_GLO_ewoco_* in Argentina
- [ ] 2020_RW_WAPOR-Ya, 2021_RWA_WAPOR-Muvu, 2021_RWA_WAPOR-Yan

There are issues with other African datasets too, but maybe crop/not_crop definition should be revised first for them.

**Comment by kvantricht** [comment]
@cbutsko I guess the results above need an updated based on the new finetuned embeddings? In this case, I'll wait for your new findings and if there's still `ref_ids` that need checking, I can look into that. It would also be cool if we can actually use that method to identify potentially corrupt datasets and iteratively clean up the training data where needed.

**Comment by cbutsko** [comment]
@kvantricht results are [here](https://vitoresearch-my.sharepoint.com/:x:/g/personal/christina_butsko_vito_be/ETkzycY-iIFIk_2oed-RfKMBEhyuL61Lp-q9uyhr4gevow?e=kQpjGL)

I marked those ref_id/country that require additional check in red or yellow, with particular comments on what needs to be checked. This was done based on country LOO performance of the best model (crop/not_crop Catboost model with finetuned Presto embeddings) and the following things:
- comparison of performance between various ref_ids in the same country
- incorporation of number of crop class points
- manual inspection of errors in QGIS

Here's a list for convenience:
```
COD 2020_GLO_ewoco*
IDN 2020_GLO_ewoco*
PRT 2018_EU_LUCAS
UGA 2020_GLO_ewoco*
ESP 2019_ES_SIGPAC-Catalunya
ITA 2018_EU_LUCAS
KAZ 2020_GLO_ewoco*
RUS 2020_GLO_ewoco*
SDN 2020_GLO_ewoco*
UKR 2020_GLO_ewoco*
MOZ 2020_GLO_ewoco*
MOZ 2021_MOZ_WAPOR
```
Also, I used the best finetuned Presto embeddings to update the confusion matrix [here](https://vitoresearch-my.sharepoint.com/:x:/g/personal/christina_butsko_vito_be/EYEvsyl8lllGpNiCc234XnYBwhoYQKJD2AvoUXViwFe4GA?e=jcaseg). Specifically, we can look at crops of interest and see how many of them have been classified as not_crop by our best cropland model. I suggest to look into those labels where confusion is beyond some _reasonable doubt_: it's understandable that maize can be confused with another crop, but when 20% of maize points are not even recognized as crop, it's suspicious (I highlighted suspicious ones in red). The following labels require particular attention:
- **oats** in Austria, Spain and Latvia (major sources of this croptype); this is a very difficult and rare class by itself (only 1.6K points), and 17% of oats points are confidently predicted as not_crop (which sets the quality ceiling for croptype model)
- **rye**; similar situation as with oats: only 1K points, and 12% of them are confidently predicted as not_crop
- **winter wheat** in USA
- **barley** in Spain 
- for **maize**, ~12-14% of not_crop predictions are happening in Belgium, Spain, Netherlands and suddenly USA. It's probably not critical for performance, as a lot of other data can compensate for that, but still notable 
- also, in Belgium ~10% of winter cereals are consistently predicted as not_crop over the years. Is it possible that this is crop that died and wasn't replanted?

**Comment by kvantricht** [comment]
Interesting analyses! Some of the findings make sense and we need to find a way to take this into account. Other results (like Belgium and LUCAS) are more surprising to me. Let's find some time to go over in more detail and also feed the information back to WENR.
